### ISSUE TITLE
moment function doesn't work with SpectrumCollection

### ISSUE THREAD

**Issue by ibusko** [astropy/specutils#772]
Results in an error like this:

`ValueError: operands could not be broadcast together with shapes (2,10000) (4,10000)`

**Comment by dhomeier** [comment]
https://github.com/astropy/specutils/blob/625e05499386e4c48467a4a141d6763e9147fcbc/specutils/analysis/moment.py#L68-L78
is not doing the right thing for `SpectrumCollection` since `spectral_axis` already has the same shape as `flux`.
Also the final subtraction does not work for non-scalar `m1` (both `Spectrum1D` with multi-D flux and `SpectrumCollection`). Should be something like this (untested and probably not working for >2D):
```python
    if len(flux.shape) > len(spectral_axis.shape):
        dispersion = np.tile(spectral_axis, [flux.shape[0], 1])

    if order == 1:
        return np.sum(flux * dispersion, axis=axis) / np.sum(flux, axis=axis)

    if order > 1:
        m0 = np.sum(flux, axis=axis)
        m1 = np.sum(flux * dispersion, axis=axis) / np.sum(flux, axis=axis)
        if len(dispersion.shape) > 1:
            m1 = np.tile(m1, [dispersion.shape[-1], 1]).T

        return np.sum(flux * (dispersion - m1)**order, axis=axis) / m0
```

**Comment by pllim** [comment]
Fixed in #781
